src/components/ConnectedTitle/ConnectedTitle.test.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { keyBy } from 'lodash';
+
+import { DataContext } from '../../contexts';
+import { identifierTypes, instanceRelationshipTypes } from '../../../test/fixtures';
+import ConnectedTitle from './ConnectedTitle';
+import renderWithIntl from '../../../test/jest/helpers/renderWithIntl';
+
+jest.mock('../InstancePlugin/InstancePlugin', () => ({ onSelect, onClose }) => {
+  return (
+    <div>
+      <div>InstancePlugin</div>
+      <button type="button" onClick={() => onSelect()}>onSelect</button>
+      <button type="button" onClick={() => onClose()}>onClose</button>
+    </div>
+  );
+});
+
+const renderConnectedTitle = (props = {}) => renderWithIntl(
+  <Router>
+    <DataContext.Provider value={{
+      contributorTypes: [],
+      identifierTypes,
+      identifierTypesById: keyBy(identifierTypes, 'id'),
+      identifierTypesByName: keyBy(identifierTypes, 'name'),
+      instanceRelationshipTypes,
+      instanceRelationshipTypesById: keyBy(identifierTypes, 'id'),
+      instanceFormats: [],
+      modesOfIssuance: [],
+      natureOfContentTerms: [],
+      tagsRecords: [],
+    }}
+    >
+      <ConnectedTitle
+        instance={{
+          title: '',
+          hrid: '',
+          identifiers: [],
+        }}
+        onSelect={() => {}}
+        titleIdKey="id"
+        {...props}
+      />
+    </DataContext.Provider>
+  </Router>
+);
+
+
+describe('ConnectedTitle', () => {
+  it('should render correct titles', () => {
+    const { queryByText } = renderConnectedTitle();
+
+    expect(queryByText('ui-inventory.precedingField.title')).toBeInTheDocument();
```

Comment:
If you render the component with intl, shouldn't it render translations values instead of ids?

Reply:
Good point, thanks! I have added translations. To be honest, I like the option with keys more if we check what exactly was rendered and should not care about how it will be translated. I see that on the folio it's about 50/50, some just do keys, some add translations, but it looks like in inventory you do this with translations, so it makes sense

Reply:
yes, I guess it makes sense to check values because we are actually testing UI and that's why we use react-testing-library, so it is necessary to check what a user sees on UI